server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/vacation/CassandraVacationDAO.java

```diff
@@ -104,11 +107,30 @@ public CassandraVacationDAO(Session session, CassandraTypesProvider cassandraTyp
                 .getZonedDateTime());
     }
 
-    private UDTValue convertToUDTValue(Optional<ZonedDateTime> zonedDateTimeOptional) {
+    private Insert createSpecificUpdate(VacationPatch vacationPatch, Insert baseInsert) {
+        return insertGeneratorPipeline
+            .apply(vacationPatch)
+            .apply(baseInsert);
+    }
+
+    public <T> Function<Insert, Insert> addValue(String field, MissableOptional<T> missableValue) {
+        return missableValue.mapNotMissingToValue(optionalValue -> getInsertInsertFunction(field, optionalValue))
+            .presentOrElse(Function.identity());
+    }
+
+    public Function<Insert, Insert> addValueZonedDateTime(String field, MissableOptional<ZonedDateTime> missableValue) {
+        return missableValue.mapNotMissingToValue(optionalValue -> getInsertInsertFunction(field, convertToUDTOptional(optionalValue)))
+            .presentOrElse(Function.identity());
+    }
+
+    private <T> Function<Insert, Insert> getInsertInsertFunction(String field, Optional<T> value) {
```

Comment:
`applyPatchForField` (you do the same thing as in `addValue` but you already get rid of the "Keep" case